volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -279,6 +279,77 @@ def get_symbol_by_address(self, wanted_sym_address):
 
         return None
 
+    def _module_flags_taints_pre_4_10_rc1(self) -> str:
+        """Convert the module's taints value to a 1-1 character mapping.
+        Relies on statically defined taints mappings in the framework.
+
+        Returns:
+            The raw taints string.
+        """
+        taints_string = ""
+        for char, taint_flag in linux_constants.TAINT_FLAGS.items():
+            if taint_flag.module and self.taints_value & taint_flag.shift:
+                taints_string += char
+
+        return taints_string
+
+    def _module_flags_taints_post_4_10_rc1(self) -> str:
+        """Convert the module's taints value to a 1-1 character mapping.
+        Relies on kernel symbol embedded taints definitions.
+
+            struct taint_flag {
+                char c_true;		/* character printed when tainted */
+                char c_false;		/* character printed when not tainted */
+                bool module;		/* also show as a per-module taint flag */
+            };
+
+        Returns:
+            The raw taints string.
+        """
+        taints_string = ""
+        for i, taint_flag in enumerate(self.taint_flags_list):
+            c_true = chr(taint_flag.c_true)
+            c_false = chr(taint_flag.c_false)
+            if taint_flag.module and (self.taints_value & (1 << i)):
+                taints_string += c_true
+            elif taint_flag.module and c_false != " ":
+                taints_string += c_false
+
+        return taints_string
+
+    def get_taints_as_plain_string(self) -> str:
+        """Convert the module's taints value to a 1-1 character mapping.
+
+        Returns:
+            The raw taints string.
+
+        Documentation:
+            - module_flags_taint kernel function
+        """
+
+        if self.taint_flags_list:
+            return self._module_flags_taints_post_4_10_rc1()
+        return self._module_flags_taints_pre_4_10_rc1()
+
+    def get_taints_parsed(self) -> List[str]:
+        """Convert the module's taints string to a 1-1 descriptor mapping.
+
+        Returns:
+            A comprehensive (user-friendly) taint descriptor list.
+
+        Documentation:
+            - module_flags_taint kernel function
+        """
+        comprehensive_taints = []
+        for c in self.get_taints_as_plain_string():
```

Comment:
Please don't use single letters in loops.  You only have to type it once, but `character` will just make it a little easier for readers of the code for years to come...